ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -747,6 +799,159 @@ export class ObjectDetailComponent implements OnInit {
     this.addedPkColumnName = columnName
   }
 
+  addCcColumn() {
+    const index = this.ccData.length;
+  
+    this.ccData.push({
+      spSno: (index + 1).toString(),
+      spConstraintName: `Constraint_name${index + 1}`,
+      spCondition: '',
+      srcSno: '',
+      srcCondition: '',
+      srcConstraintName: '',
+      deleteIndex: `cc${index + 1}`,
+    });
+  
+    this.setCCRows();
+  }
+    
+  dropCc(element: any) {
+    const index = this.ccData.findIndex(item => item.deleteIndex === element.value.deleteIndex);
+    if (index !== -1) {
+      this.ccData.splice(index, 1);
+    }
+    this.setCCRows();
+  }
+
+  setCCRows() {
+    this.ccArray = this.fb.array([]);
+    const srcArr: ICcTabData[] = [];
+    const spArr: ICcTabData[] = [];
+  
+    // Populate srcArr and spArr
+    this.ccData.forEach((cc, index) => {
+      const baseObject : ICcTabData = {
+        srcSno: cc.srcSno,
+        srcConstraintName: cc.srcConstraintName,
+        srcCondition: cc.srcCondition,
+        spSno: `${index + 1}`,
+        spConstraintName: cc.spConstraintName,
+        spCondition: cc.spCondition,
+        deleteIndex: cc.deleteIndex,
+      };
+      srcArr.push(baseObject);
+  
+      if (cc.spConstraintName !== '') {
+        spArr.push(baseObject);
+      }
+    });
+  
+    const createFormGroup = (data : ICcTabData) => new FormGroup({
+      srcSno: new FormControl(data.srcSno || ''),
+      srcConstraintName: new FormControl(data.srcConstraintName || ''),
+      srcCondition: new FormControl(data.srcCondition || ''),
+      spSno: new FormControl(data.spSno || ''),
+      spConstraintName: new FormControl(data.spConstraintName || '', [
+        Validators.required,
+        Validators.pattern('^[a-zA-Z_][a-zA-Z0-9/_]*[a-zA-Z0-9]?'),
+      ]),
+      spCondition: new FormControl(data.spCondition || '', [
+        Validators.required,
+        this.checkCondition(),
+      ]),
+      deleteIndex: new FormControl(data.deleteIndex),
+    });
+
+    for (let i = 0; i < Math.min(srcArr.length, spArr.length); i++) {
+      this.ccArray.push(createFormGroup(srcArr[i]));
```

Comment:
Can you please explain what is happening in these three for loops?

Reply:
srcArr and spArr might not have the same number of elements. to insure no element left to be populated